cou/commands.py

```diff
@@ -0,0 +1,343 @@
+        help="For more information about an upgrade group, run 'cou run <upgrade-group> -h'.",
+    )
+    run_subparser.add_parser(
+        "control-plane",
+        description="Run upgrade for the control-plane components.",
+        help="Run upgrade for the control-plane components.",
+        usage="cou plan control-plane [options]",
+        parents=[subcommand_common_opts_parser, run_args_parser],
+        formatter_class=CapitalizeHelpFormatter,
+    )
+    run_subparser.add_parser(
+        "data-plane",
+        description="Run upgrade for the data-plane components. This is possible only if "
+        "control-plane has been fully upgrade. Otherwise an error will be thrown.",
+        help="Run upgrade for the data-plane components. This is possible only if "
+        "control-plane has been fully upgrade. Otherwise an error will be thrown.",
+        usage="cou plan data-plane [options]",
+        parents=[
+            subcommand_common_opts_parser,
+            dp_parser,
+            run_args_parser,
+        ],
+        formatter_class=CapitalizeHelpFormatter,
+    )
+
+
+def create_subparsers(parser: argparse.ArgumentParser) -> argparse._SubParsersAction:
+    """Create and configure subparsers.
+
+    :param parsers: the top level parser to create subparsers for,
+    :type parsers: argparse.ArgumentParser
+    :return subparsers: configured subparsers
+    :type subparsers: argparse.ArgumentParser
+    """
+    # Configure subcommand parser
+    subparsers = parser.add_subparsers(
+        title="Commands",
+        dest="command",
+        help="For more information about a command, run 'cou help <command>'.",
+    )
+
+    # Arg parser for "cou help" sub-command
+    help_parser = subparsers.add_parser(
+        "help",
+        usage="cou help [command]",
+    )
+    help_parser.add_argument(
+        "subcommand",
+        nargs="?",
+        choices=["plan", "run", "all"],
+        default="all",
+        type=str,
+        help="A sub-command to get information of.",
+    )
+
+    subcommand_common_opts_parser = get_subcommand_common_opts_parser()
+    dp_parser = get_dataplane_common_opts_parser()
+    create_plan_subparser(subparsers, subcommand_common_opts_parser, dp_parser)
+    create_run_subparser(subparsers, subcommand_common_opts_parser, dp_parser)
+
+    return subparsers
+
+
+def parse_args() -> tuple[argparse.ArgumentParser, argparse._SubParsersAction]:
+    """Parse cli arguments.
+
+    :return: Arguments parser.
+    :rtype: argparse.ArgumentParser
+    :return: Arguments subparsers.
+    :rtype: argparse._SubParsersAction
+    """
+    # Configure top level argparser and its options
+    parser = argparse.ArgumentParser(
+        description="Charmed OpenStack Upgrader (cou) is an application to upgrade a Canonical "
+        "distribution of Charmed OpenStack. The application auto-detects the version of the "
+        "running cloud and will propose an upgrade to the next available version.",
```

Comment:
This line is very long, you may need multiple `\n`s.
```suggestion
        description="Charmed OpenStack Upgrader (cou) is an application to upgrade a Canonical "
        "distribution of Charmed OpenStack.\nThe application auto-detects the version of the "
        "running cloud and will propose an upgrade to the next available version.",
```